### PR TITLE
trying to avoid split brains for 2n nodes configurations

### PR DESCRIPTION
**Pull request by benoitvidis**
## What does this PR do ?

This PR tries to mitigate the risk of split brain in case of network congestion on a 2n cluster.

The idea is to check on each heartbeat if the node was rejected by other nodes and if so, to attempt to join again.

### How should this be manually tested?

Reproducing the situation where the split brain occurs is a problem and I hardly managed to do it on my system.

### CODE REVIEW

File: docker-compose/images/kuzzle/Dockerfile [hunk 1]
```diff
@@ -4,9 +4,11 @@ RUN  set -x \
   \
   && npm set user 0 \
   && npm set unsafe-perm true \
+  && npm install -g pm2 \
```
Comment (scottinet) [inline docker-compose/images/kuzzle/Dockerfile]: Why do we need to add pm2 and git to the cluster stack?

File: lib/node.js [hunk 2]
```diff
@@ -273,7 +273,29 @@ class Node {
...
+          const member = JSON.parse(serialized);
+
+          if (member.router === this.config.bindings.router.href) {
+            found = true;
+            break;
+          }
+        }
+
+        if (!found) {
+          return this.join();
```
Comment (scottinet) [inline lib/node.js]: What if joining fails? Shouldn't the node stop accepting requests and enter a reconnection loop, or something?

File: docker-compose/images/kuzzle/Dockerfile (hunk 1 above)
Comment (benoitvidis) [reply docker-compose/images/kuzzle/Dockerfile]: The image is based on kuzzle's one (not core-dev) to let developers focus on the cluster and not have to worry about mounting kuzzle.
The last image update removed git and pm2, which are required during development, respectively for npm install and for hot-reload.

File: lib/node.js (hunk 2 above)
Comment (benoitvidis) [reply lib/node.js]: We already have a loop. This code is triggered on each heartbeat, every 5s by default.
About rejecting requests, ideallly the cluster health check should  be KO till the node is not joined (which may not be the case if we have a 2n cluster setup and an actual split brain).
If the health check is down, the reverse proxy should be configured to exclude the node from the pool.

**Pull request merged**